backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java

```diff
@@ -182,4 +200,63 @@ public boolean checkTransitionEdgeTarget(Element source, Element target) {
         }
         return false;
     }
+
+    /**
+     * Succession edge label getter service.
+     *
+     * @param succession
+     *            The succession edge
+     * @return the label of the given {@link Succession}
+     */
+    public String getSuccessionLabel(Succession succession) {
+        String resultLabel = "";
+        var guardExpressions = succession.getGuardExpression();
+        if (!guardExpressions.isEmpty()) {
+            String guardLabel = this.getGuardExpressionsDefaultDirectEditLabel(guardExpressions);
+            if (!guardLabel.isBlank()) {
+                resultLabel += "[" + guardLabel + "]";
+            }
+        }
+        return resultLabel;
+    }
+
+    private String getElementsDefaultInitialDirectEditLabel(EList<? extends Element> elements, BinaryOperator<String> reduceOperator) {
+        return elements.stream()
+                .map(action -> {
+                    if (action instanceof AcceptActionUsage aau) {
+                        return this.viewLabelService.getDefaultInitialDirectEditLabel(aau.getPayloadParameter());
+                    }
+                    return this.viewLabelService.getDefaultInitialDirectEditLabel(action);
+                })
+                .reduce(reduceOperator)
+                .orElse("");
+    }
+
+    private String getGuardExpressionsDefaultDirectEditLabel(EList<Expression> guardExpressions) {
+        String guardLabel;
+        guardLabel = this.getElementsDefaultInitialDirectEditLabel(guardExpressions, (a, b) -> {
```

Comment:
Please do not split declaration and affect of `guardLabel `.

Reply:
Done